### PR TITLE
Fix ruff violation in coordinates subpackage

### PR DESCRIPTION
**Pull request by ecalifornica**
### Description

This pull request is to address the ruff lint violation 'superfluous-else-raise' (RET506) in the `coordinates` sub-package by eliminating unnecessary else blocks after raise statements. I make no claim as to the validity of this violation, I chose it because it was suggested in #14818.

> What it does
> Checks for else statements with a raise statement in the preceding if block.
> 
> Why is this bad?
> The else statement is not needed as the raise statement will always break out of the current scope. Removing the else will reduce nesting and make the code more readable.
> - https://docs.astral.sh/ruff/rules/superfluous-else-raise/

This PR also updates the ruff temporary configuration to prevent similar future issues in this sub-package.

> ...some of the rules have a large number of violations ... in such cases it is preferable to address the rules one sub-package per pull request.
> 
> If you are editing a single sub-package and the rule is in the global ignore list then remove it from there and add it to the ignore lists of sub-packages that are violating that rule, but not to the list of the sub-package you will be updating.
> - #14818 

&nbsp;

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

**Review by nstarman — COMMENTED** [review]
Let's get a coordinates maintainer to give final approval. But this LGTM. I like to see the control flow simplified! Thanks @ecalifornica

File: astropy/coordinates/angles/formats.py
Comment (mhvk) [inline astropy/coordinates/angles/formats.py:314]: Bit of a fly-by comment, but this seems less clear than what was there before

File: astropy/coordinates/earth.py
Comment (mhvk) [inline astropy/coordinates/earth.py:86]: Same

**Review by mhvk — COMMENTED** [review]
Mostly for the better, but not sure why
```
if <something>:
    raise <One Error>
else:
   raise <Another Error>
```
would be considered bad form. Is this the same `ruff` rule? If it is a separate one, I suggest not enforcing that. If it is the same rule, probably not worth special-casing these.

**Comment by ecalifornica** [comment]
> Mostly for the better, but not sure why
> 
> ```
> if <something>:
>     raise <One Error>
> else:
>    raise <Another Error>
> ```
> 
> would be considered bad form. Is this the same `ruff` rule? If it is a separate one, I suggest not enforcing that. If it is the same rule, probably not worth special-casing these.

Yes, this is the same ruff rule.

**Review by eerovaher — APPROVED** [review]
I think this would be good to merge, but it would be best if @mhvk would also approve.

File: astropy/coordinates/angles/formats.py
Comment (mhvk) [reply astropy/coordinates/angles/formats.py:314]: My tendency would be to write this as
```
raise ValueError(
    f"{str(e) or "syntax error"} parsing angle {angle!r}"
) from e
```

File: astropy/coordinates/earth.py
Comment (mhvk) [reply astropy/coordinates/earth.py:86]: Here, I'd avoid the ugliness with
```
msg = (
    "connection timed out"
    if isinstance(e.reason, socket.timeout)
    else e.reason
)
raise NameResolveError(err_str.format(msg=msg)) from e
```

Should add that I really don't know why we bother with prettifying exception messages in so much detail...

**Review by mhvk — APPROVED** [review]
OK, some suggestions for removing the things I dislike. But up to you, @ecalifornica, if you want to implement them - this is really a bit of a detail. I also approve as is.

**Comment by ecalifornica** [comment]
> OK, some suggestions for removing the things I dislike. But up to you, @ecalifornica, if you want to implement them - this is really a bit of a detail. I also approve as is.

Made these changes.

File: astropy/coordinates/angles/core.py
Comment (astrofrog) [inline astropy/coordinates/angles/core.py:164]: What was wrong with the way the code was before?

Comment (ecalifornica) [reply astropy/coordinates/angles/core.py:164]: Looks like this linter rule doesn't like an `else` statement after a `raise`. See: https://docs.astral.sh/ruff/rules/superfluous-else-raise/

**Review by mhvk — APPROVED** [review]
Looks good to me, thanks for making the extra changes!

**Pull request merged**
